internal/security/config/command/proxy/deluser/command.go

```diff
@@ -0,0 +1,99 @@
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/security/config/interfaces"
+	"github.com/edgexfoundry/edgex-go/internal/security/proxy/config"
+	"github.com/edgexfoundry/edgex-go/internal/security/secretstoreclient"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+const (
+	CommandName string = "deluser"
+)
+
+type cmd struct {
+	loggingClient logger.LoggingClient
+	client        internal.HttpCaller
+	configuration *config.ConfigurationStruct
+	tokenType     string
+	username      string
+}
+
+func NewCommand(
+	lc logger.LoggingClient,
+	configuration *config.ConfigurationStruct,
+	args []string) (interfaces.Command, error) {
+
+	cmd := cmd{
+		loggingClient: lc,
+		client:        secretstoreclient.NewRequestor(lc).Insecure(),
+		configuration: configuration,
+	}
+	var dummy string
+
+	flagSet := flag.NewFlagSet(CommandName, flag.ContinueOnError)
+	flagSet.StringVar(&dummy, "confdir", "", "") // handled by bootstrap; duplicated here to prevent arg parsing errors
+
+	flagSet.StringVar(&cmd.tokenType, "token-type", "", "Type of token to create: jwt or oauth2")
+	flagSet.StringVar(&cmd.username, "user", "", "Username of the user to delete")
+
+	err := flagSet.Parse(args)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse command: %s: %w", strings.Join(args, " "), err)
+	}
+	if cmd.tokenType == "" {
+		return nil, fmt.Errorf("%s proxy deluser: argument --token-type is required", os.Args[0])
+	}
+	if cmd.tokenType != interfaces.JwtTokenType && cmd.tokenType != interfaces.OAuth2TokenType {
+		return nil, fmt.Errorf("%s proxy deluser: argument --token-type must be either 'jwt' or 'oauth2'", os.Args[0])
+	}
+	if cmd.username == "" {
+		return nil, fmt.Errorf("%s proxy deluser: argument --user is required", os.Args[0])
+	}
+
+	return &cmd, err
+}
+
+func (c *cmd) Execute() (int, error) {
+	// Delete Kong consumer
+	// https://docs.konghq.com/2.1.x/admin-api/#delete-consumer
+
+	kongURL := strings.Join([]string{c.configuration.KongURL.GetProxyBaseURL(), "consumers", c.username}, "/")
+	c.loggingClient.Info(fmt.Sprintf("deleting consumer (user) on the endpoint of %s", kongURL))
+
+	req, err := http.NewRequest(http.MethodDelete, kongURL, nil)
+	if err != nil {
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Failed to prepare delete consumer request %s: %w", c.username, err)
+	}
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Failed to send delete consumer request %s: %w", c.username, err)
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusNoContent:
+		c.loggingClient.Info(fmt.Sprintf("deleted consumer (user) '%s'", c.username))
+	default:
+		responseBody, _ := ioutil.ReadAll(resp.Body)
+		c.loggingClient.Error(fmt.Sprintf("Error response: %s", responseBody))
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Delete consumer request failed with code: %d", resp.StatusCode)
+	}
+
```

Comment:
What happened to deleting JWT and OAuth2?

Reply:
It turns out that the only difference between the two was the log message.

As such, ExecuteDeleteJwt(), ExecuteDeleteOauth2() were removed (since with the silent interface there is no longer a need to be chatty).  Also both of these functions called ExecuteDeleteConsumer() which does a cascading delete in the Kong DB to remove the user and their secrets.  Thus, the both of ExecuteDeleteConsumer() was rolled into Execute() in its entirety.

There is now no longer any discernable difference between deleting a JWT-based user and an Oauth2-based user.